From bafbc4a6aa2c9cf18803444b2fb426a91bb3fbd4 Mon Sep 17 00:00:00 2001
From: Mohammed Danish <shaan3@gmail.com>
Date: Fri, 7 Jul 2017 11:02:39 +0530
Subject: [PATCH 2/6] Forward the Stereo 3D metadata to be read later

---
 src/input/decoder.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/input/decoder.c b/src/input/decoder.c
index dac313737a..8404bcc95b 100644
--- a/src/input/decoder.c
+++ b/src/input/decoder.c
@@ -418,8 +418,7 @@ static int vout_update_format( decoder_t *p_dec )
      || p_dec->fmt_out.i_codec != p_owner->fmt.video.i_chroma
      || (int64_t)p_dec->fmt_out.video.i_sar_num * p_owner->fmt.video.i_sar_den !=
         (int64_t)p_dec->fmt_out.video.i_sar_den * p_owner->fmt.video.i_sar_num ||
-        p_dec->fmt_out.video.orientation != p_owner->fmt.video.orientation ||
-        p_dec->fmt_out.video.multiview_mode != p_owner->fmt.video.multiview_mode )
+        p_dec->fmt_out.video.orientation != p_owner->fmt.video.orientation )
     {
         vout_thread_t *p_vout;
 
@@ -533,14 +532,15 @@ static int vout_update_format( decoder_t *p_dec )
             return -1;
         }
     }
-
-    if ( memcmp( &p_dec->fmt_out.video.mastering,
+    else if ( memcmp( &p_dec->fmt_out.video.mastering,
                  &p_owner->fmt.video.mastering,
                  sizeof(p_owner->fmt.video.mastering)) ||
          p_dec->fmt_out.video.lighting.MaxCLL !=
          p_owner->fmt.video.lighting.MaxCLL ||
          p_dec->fmt_out.video.lighting.MaxFALL !=
-         p_owner->fmt.video.lighting.MaxFALL)
+         p_owner->fmt.video.lighting.MaxFALL ||
+         p_dec->fmt_out.video.multiview_mode !=
+         p_owner->fmt.video.multiview_mode )
     {
         /* the format has changed but we don't need a new vout */
         vlc_mutex_lock( &p_owner->lock );
-- 
2.12.2.windows.2

